templates/cisco_nxos_show_environment.textfsm

```diff
@@ -0,0 +1,55 @@
+Value POWER_SUPPLY (\d+)
+Value POWER_SUPPLY_MODEL (\S+)
+Value POWER_SUPPLY_OUTPUT (\d+)
+Value POWER_SUPPLY_INPUT (\d+)
+Value POWER_SUPPLY_CAPACITY (\d+)
+Value POWER_SUPPLY_STATUS (\w+)
+Value POWER_SUPPLY_VOLTAGE (\d+)
+Value FAN (\S+)
+Value FAN_STATUS (\S+)
+Value TEMPERATURE_MODULE (\d+)
+Value TEMPERATURE_SENSOR (\S+)
+#([^ \t\r\n\f])
+Value TEMPERATURE_MAJOR_THRESH (\d+)
+Value TEMPERATURE_MINOR_THRESH (\d+)
+Value TEMPERATURE_CURRENT (\d+)
+Value TEMPERATURE_STATUS (\S+)
+
+Start
```

Comment:
```suggestion
Start
  # Note: 2020-12 this template is broken, mashing 4 tables into one.
  ^Power\s+Supply:
  ^Voltage
  ^Power\s+Actual\s+Total -> Power
  ^Power\s+Actual\s+Actual\s+Total -> Power
  ^Fan:\s*$$ -> Fan
  ^Temperature:\s*$$ -> Temperature
  #^Fan\s+Model\s+Hw\s+(Direction\s+)?Status -> Fan

Power
  # Capture the remaining headers
  ^Supply\s+Model\s+Output\s+Capacity\s+Status
  ^Supply\s+Model\s+Output\s+Input\s+Capacity\s+Status
  ^\s+\(Watts\s\)
  ^-+
  # Start capturing the data for the two different types of outputs
  ^${POWER_SUPPLY}\s+${POWER_SUPPLY_MODEL}\s+${POWER_SUPPLY_OUTPUT}\s+\w+\s+${POWER_SUPPLY_CAPACITY}\s+\w+\s+${POWER_SUPPLY_STATUS} -> Record
  ^${POWER_SUPPLY}\s+${POWER_SUPPLY_MODEL}\s+${POWER_SUPPLY_OUTPUT}\s+\w+\s+${POWER_SUPPLY_INPUT}\s+\w+\s+${POWER_SUPPLY_CAPACITY}\s+\w+\s+${POWER_SUPPLY_STATUS} -> Record
  ^Power\s+Usage\s+Summary: -> Start
  ^\s*$$ -> Start
  ^. -> Error
```

Sorry this template is pretty complicated. Above are some changes I would personally make for readability and transferring states closer to the data that will be captured.